src/plugins/data_source/server/client/client_config.ts

```diff
@@ -18,14 +28,70 @@ export function parseClientOptions(
   endpoint: string,
   registeredSchema: any[]
 ): ClientOptions {
+  const sslConfig: DataSourceSSLConfigOptions = {
+    requestCert: true,
+    rejectUnauthorized: true,
+  };
+
+  if (config.ssl) {
+    const verificationMode = config.ssl.verificationMode;
+    switch (verificationMode) {
+      case 'none':
+        sslConfig.rejectUnauthorized = false;
+        break;
+      case 'certificate':
+        sslConfig.rejectUnauthorized = true;
+
+        // by default, NodeJS is checking the server identify
+        sslConfig.checkServerIdentity = () => undefined;
+        break;
+      case 'full':
+        sslConfig.rejectUnauthorized = true;
+        break;
+      default:
+        throw new Error(`Unknown ssl verificationMode: ${verificationMode}`);
+    }
+
+    const { certificateAuthorities } = readCertificateAuthorities(config);
+
+    sslConfig.ca = certificateAuthorities || [];
+  }
+
   const clientOptions: ClientOptions = {
     node: endpoint,
-    ssl: {
-      requestCert: true,
-      rejectUnauthorized: true,
-    },
+    ssl: sslConfig,
     plugins: registeredSchema,
   };
 
   return clientOptions;
 }
+
+export const readCertificateAuthorities = (rawConfig: any) => {
```

Comment:
Looks like we only need `rawConfig.ssl.certificateAuthorities` here.
Maybe we can only pass that value as the parameter of this function instead of pass the whole config with an any type. What do you think?

Reply:
In the future, this method could be updated to add more TLS configuration like is done with the client for [default opensearch](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/opensearch/opensearch_config.ts#L364-L425). Since only certificateAuthorities is supported for the datasource opensearch client, I will update this to only consume certificateAuthorities as a parameter.

Reply:
Changed this to consume only the certificateAuthorities param.